lib/api/core.dart

```diff
@@ -10,6 +10,24 @@ import '../model/binding.dart';
 import '../model/localizations.dart';
 import 'exception.dart';
 
+/// The Zulip Server version below which we should refuse to connect.
+///
+/// When updating this, also update [kMinSupportedZulipFeatureLevel]
+/// and the README.
+const kMinSupportedZulipVersion = '4.0';
+
+/// The Zulip feature level reserved for the [kMinSupportedZulipVersion] release.
+///
+/// For this value, see the API changelog:
+///   https://zulip.com/api/changelog
+const kMinSupportedZulipFeatureLevel = 65;
+
+/// The doc stating our oldest supported server version.
+// TODO: Instead, link to new Help Center doc once we have it:
+//   https://github.com/zulip/zulip/issues/23842
+final kServerSupportDocUrl = Uri.parse(
+  'https://zulip.readthedocs.io/en/latest/overview/release-lifecycle.html#compatibility-and-upgrading');
```

Comment:
```suggestion
  'https://zulip.readthedocs.io/en/latest/overview/release-lifecycle.html#client-apps');
```

The other URL's fragment doesn't seem to exist on that page now.

(It's fine, the page still loads, but the user doesn't get taken to any particular spot on it.)